Helper/AlgoliaHelper.php

```diff
@@ -119,16 +115,17 @@ protected function addAlgoliaUserAgent(): void
     }
 
     /**
-     * @return SearchClient
+     * @return SearchClient|null
      * @throws AlgoliaException
      */
-    public function getClient(): SearchClient
+    public function getClient(): ?SearchClient
     {
         if (!isset($this->clients[$this->getStoreId()])) {
             $this->createClient();
+            $this->addAlgoliaUserAgent();
```

Comment:
Won't this get called redundantly on the static class method? Or do we care?

Reply:
Added a boolean to prevent that